text/0039-_formatted-field-behavior.md

```diff
@@ -0,0 +1,136 @@
+
+Given a document made of two fields `title` and `poster`
+```json
+{
+    "title": "Prince Avalance",
+    "poster": "https://image.tmdb.org/t/p/w1280/3KHiQt54usbHyIjLIMzaDAoIJNK.jpg"
+}
+```
+
+Given these search parameters:
+```json
+{
+    "q": "prince",
+    "attributesToHighlight": ["title"]
+}
+```
+
+As a user i expect to have this response:
+
+```json
+{
+    "hits": [
+        {
+            "title": "Prince Avalanche",
+            "poster": "https://image.tmdb.org/t/p/w1280/3KHiQt54usbHyIjLIMzaDAoIJNK.jpg",
+            "_formatted": {
+                "title": "<em>Prince</em> Avalance"
+            }
+        }
+    ]
+}
+```
+
+but i get:
+```json
+{
+    "hits": [
+        {
+            "title": "Prince Avalanche",
+            "poster": "https://image.tmdb.org/t/p/w1280/3KHiQt54usbHyIjLIMzaDAoIJNK.jpg",
+            "_formatted": {
+                "title": "<em>Prince</em> Avalance",
+                "poster": "https://image.tmdb.org/t/p/w1280/3KHiQt54usbHyIjLIMzaDAoIJNK.jpg"
+            }
+        }
+    ]
+}
+```
+
+`_formatted` is filled with all the `displayedAttributes` despite the fact that the user only ask for one specific field.
+
+This behaviour will be fixed in the `0.21` release in order to only get the fields set in `attributesToHighlight` and `attributesToCrop` in `_formatted`.
+
+#### MeiliSearch Behavior (0.21)
+
+As in the `0.20`, the `_formatted` fields appears in the search response as soon as the `attributesToHighlight` and/or `attributesToCrop` parameters are sent with an empty value.
+
+E.g. `"attributesToHighlight": `
+
+As in the `0.20`, `_formatted` is also filled with all the `displayedAttributes` despite the fact that the user only ask for one specific field in `attributesForHighlight` or `attributesToCrop`.
+
+The `_formatted` field appears in the search response as soon as the `attributesToHighlight` and/or `attributesToCrop` parameters are sent as a query parameter and filled with an existent parameter.
+
+The next rules clarify the behavior of `_formatted` parameter given `attributesToHighlight` and `attributesToCrop` states and combination.
+
+✅  `_formatted` response parameter **SHOULD NOT** be returned in the given cases:
+
+- `attributesToHighlight` is set but is empty AND `attributesToCrop` is set but is empty.
+- `attributesToHighlight` is set but is empty AND `attributesToCrop` isn't set.
+- `attributesToHighlight` is set but only contains a non-existent field AND `attributesToCrop` is set but only contains a non-existent field.
+- `attributesToHighlight` is set but only contains a non-existent field AND (`attributesToCrop` is not set OR `attributesToCrop` is set but is empty).
+
+✅ `_formatted` response parameter **SHOULD** be returned in any cases of:
+
+- `attributesToHighlight` is set with `*`. When `attributesToHighlight` is not set, the engine will use `*` as default parameter. So each `displayedAttributes` will be in the `_formated` result in that specific case.
```

Comment:
> `attributesToHighlight` is set with `*`. When `attributesToHighlight` is not set, the engine will use `*` as default parameter.

If `attributesToHighlight` is set, it means the users don't want highlight, so `*` should not be the default behavior, right?